src/geff/testing/_utils.py

```diff
@@ -58,3 +65,70 @@ def nx_is_equal(g1: nx.Graph, g2: nx.Graph) -> bool:
         return True
     else:
         return False
+
+
+def check_equiv_geff(store_a: StoreLike, store_b: StoreLike) -> None:
+    """This function compares two geffs, typically a starting fixture geff with
+    the output of an implementation.
+
+    This tests focuses on maintaining shape and dtype consistency. It does not
+    assert element wise equality. store_a is assumed to be the "correct" geff.
+
+    Missing arrays are not required to be present in both a and b because we allow
+    missing arrays where all values are present. We raise a warning if we see this.
+
+    Args:
+        store_a (str | Path | zarr store): str/Path/store for a geff zarr
+        store_b (str | Path | zarr store): str/Path/store for a second geff zarr
+    """
+
+    # Run validation first so that we don't hit issues with basic structure
+    validate_structure(store_a)
+    validate_structure(store_b)
+
+    for graph_group in [_path.NODES, _path.EDGES]:
+        ga = expect_group(open_storelike(store_a), graph_group)
+        gb = expect_group(open_storelike(store_b), graph_group)
+
+        # Check ids
+        ids_a = expect_array(ga, _path.IDS)
+        ids_b = expect_array(gb, _path.IDS)
+        if (a_shape := ids_a.shape) != (b_shape := ids_b.shape):
+            raise ValueError(f"{graph_group} ids shape: a {a_shape} does not match b {b_shape}")
+
+        # Check that properties in each geff are the same
+        a_props = set(expect_group(ga, _path.PROPS))
+        b_props = set(expect_group(gb, _path.PROPS))
+        if a_props != b_props:
+            raise ValueError(
+                f"{graph_group} properties: a ({a_props}) does not match b ({b_props})"
+            )
+
+        # Check shape and dtype of each prop
+        for prop in a_props:
+            a_missing = "missing" in expect_group(ga, f"{_path.PROPS}/{prop}")
+            b_missing = "missing" in expect_group(gb, f"{_path.PROPS}/{prop}")
```

Comment:
not a suggest for this PR, but just a thought... 

I can see a lot of times where we are re-duplicating things across the library.  (i.e. after validation has been run, re-running expect etc...).  I wonder whether it might be useful to have a `GeffFile` class (something like `TiffFile` or `ND2File`) with dedicated methods (like `nodes() -> zarr.Group` and `edges() -> zarr.Group` and `node_ids() -> zarr.Array`).  These would be both "convenient" (in the sense that you don't need to use strings or imports from `_path` to access canonical parts of a geff file) and also useful for typing, since you can validate once and cast everywhere else.  

I'll play with it...